GVRf/Framework/src/org/gearvrf/utility/Preconditions.java

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.utility;
+
+public final class Preconditions {
+    
+    /**
+     * The goal of this class is to improve the readability and to provide some common checking method to improve 
+     * the robustness of GVRF. But in some cases, this may be a significant performance cost. Try to use and further
+     * extend this class carefully with concern on performance issue.
+     * 
+     * Actually, one purpose of this class may help us avoid from using GVRJniException which is not a proper way to 
+     * throw/simulate an exception with JNI crash.
+     */
+    
+
+    public static final String INPUT_TYPE_ARRAY = "Input array";
```

Comment:
1. `public` members should have JavaDoc.
2. These strings really don't belong here: if the same code is used in a new way, someone would have to come here and change the set of strings. It would be better if each `checkX()` method took a `String parameterName`, so the exception message could refer to the name in the calling code.